### ISSUE TITLE
Environment variable config not handled by reactive messaging Kafka connector

### ISSUE THREAD

**Issue by Azquelt** [OpenLiberty/open-liberty#10575]
Reported by an internal user

**Describe the bug**

When defining a config property like
`MP_MESSAGING_CONNECTOR_LIBERTY_KAFKA_BOOTSTRAP_SERVERS=myserver.local`, this does not result in a `bootstrap.servers=myserver.local` property being added to the config map passed to `KafkaProducer` or `KafkaConsumer`.

**Additional context**

I think this occurs because we retrieve all configured values which aren't known by our connector and pass them through to Kafka. This means that we get the key name as `BOOTSTRAP_SERVERS` instead of the expected `bootstrap.servers`.

More investigation is needed to discover whether the smallrye reactive messaging provider will find all of the names which might refer to our connector or whether they're already missing from our list by the time we receive the config object.

We're investigating a possible workaround of defining all desired properties in microprofile-config.properties so that we pick up the correct property name from microprofile-config.properties and then get the correct value from environment variables.

**Comment by scottkurz** [comment]
I hit this on 20.0.0.3 in working on the kafka template for the java-openliberty stack (source [here](https://github.com/appsody/stacks/tree/master/incubator/java-openliberty/templates/kafka)). 

I learned that as long as I had an entry in **microprofile-config.properties** like:
>  mp.messaging.connector.liberty-kafka.bootstrap.servers=localhost:9093
I could override with an env var:
> MP_MESSAGING_CONNECTOR_LIBERTY_KAFKA_BOOTSTRAP_SERVERS=myserver:9092

The problem is if I remove the **mp.messaging.connector.liberty-kafka.bootstrap.servers** property from **microprofile-config.properties**...then having the above env var set has no effect.  I see:

> 2020-04-30 20:17:16 INFO  ConsumerConfig:347 - ConsumerConfig values:
>         allow.auto.create.topics = true
>         auto.commit.interval.ms = 5000
>         auto.offset.reset = latest
>         bootstrap.servers = []

So basically having the property in **microprofile-config.properties** is a de facto requirement now in order to use the env var override (it overrides because it has the higher ordinal value).

**Comment by jwalcorn** [comment]
Any news on this?  I'm having to put bogus values in my microprofile-config.properties, so that the env vars I feed in from my Kube config map/secret will work.  Not a big deal, but seems like a hack to have to define fake values in order to get MP Config to accept my env vars
